Docker/UsefulCommands.md

```diff
@@ -0,0 +1,141 @@
+Build an image tagged `my-org:my-image` where the Dockerfile can be found at
+`/tmp/Dockerfile`.
+
+```sh
+docker build -t my-org:my-image -f /tmp/Dockerfile
+```
+
+### Useful flags
+- `--file -f` Path where to find the Dockerfile
+- `--force-rm` Always remove intermediate containers
+- `--no-cache` Do not use cache when building the image
+- `--rm` Remove intermediate containers after a successful build (this is
+`true`) by default
+- `--tag -t` Name and optionally a tag in the ‘name:tag’ format
+
+
+## [`docker exec`](https://docs.docker.com/engine/reference/commandline/exec/)
+Execute a command inside a **running** container.
+
+```sh
+docker exec [CONTAINER ID]
+```
+
+### Example
+```sh
+docker exec [CONTAINER ID] touch /tmp/exec_works
+```
+
+### Useful flags
+- `--detach -d` Detached mode: run command in the background
+- `-it` This will not make the container you started shut down immediately, as
+it will create a pseudo-TTY session (`-t`) and keep STDIN open (`-i`)
+
+
+## [`docker images`](https://docs.docker.com/engine/reference/commandline/images/)
+List all downloaded/created images.
+
+```sh
+docker images
+```
+
+### Useful flags
+- `-q` Only show numeric IDs
+
+
+## [`docker inspect`](https://docs.docker.com/engine/reference/commandline/inspect)
+Shows all the info of a container.
+
+```sh
+docker inspect [CONTAINER ID]
+```
+
+
+## [`docker logs`](https://docs.docker.com/engine/reference/commandline/logs/)
+Gets logs from container.
+
+```sh
+docker logs [CONTAINER ID]
+```
+
+### Useful flags
+- `--details` Log extra details
+- `--follow -f` Follow log output
+- `--timestamps -t` Show timestamps
+
+
+## [`docker ps`](https://docs.docker.com/engine/reference/commandline/ps/)
+Shows information about all running containers.
+
+```sh
+docker ps
+```
+
+### Useful flags
+- `--all -a` Show all containers (default shows just running)
+- `--filter -f` Filter output based on conditions provided, `docker ps -f="name="example"`
+- `--quiet -q` Only display numeric IDs
+
+
+## [`docker rmi`](https://docs.docker.com/engine/reference/commandline/rmi/)
+Remove one or more images.
+
+```sh
+docker rmi [IMAGE ID]
+```
+
+### Useful flags
+- `--force -f` Force removal of the image
+
+
+## [`docker run`](https://docs.docker.com/engine/reference/commandline/run/)
+Creates and starts a container in one operation. Could be used to execute a
+single command as well as start a long-running container.
+
+Example:
+
+```sh
+docker run -it ubuntu:latest /bin/bash
+```
+
+This will start a ubuntu container with the entrypoint `/bin/bash`. Note that
+if you do not have the `ubuntu` image downloaded it will download it before
+running it.
+
+### Useful flags
+- `-it` This will not make the container you started shut down immediately, as
+it will create a pseudo-TTY session (`-t`) and keep STDIN open (`-i`)
+- `--rm` Automatically remove the container when it exit. Otherwise it will
+live on and take up space (see all images running `docker images`)
```

Comment:
I am not sure if usage of "live on" is correct, since the container is in a "stopped" state.

I would phrase it as following:
- `--rm` Automatically remove the container when it exits. Otherwise it will be visible when you run `docker ps -a`.

Reply:
Even that seems a bit odd without the proper context.

Reply:
Yeah I agree, it could be phrased better. I'll come up with something.